### DISCUSSION TITLE
Access System Text Size and react to changes on Windows 11

### DISCUSSION THREAD

**Discussion by cricketthomas** [AvaloniaUI/Avalonia#14579] [Q&A]
On Windows 11 there is the text size setting that changes the font of every application, and I haven't been able to find if there's a dynamic resource for fonts that would adapt to the text size changes. 

I tested it with a default template app and there the app doesn't seem to react. 
I am not sure there is something additional that needs to be enabled for this to happen.

Screenshot:
![image](https://github.com/AvaloniaUI/Avalonia/assets/15821271/85369fb4-ccf4-4b96-a810-edb9af919f03)

Thanks

**Comment by maxkatz6** [comment]
It should be done in two steps at least:
1. Add new API in PlatformSettings interface. Implement on following platforms.
2. Figure out how to handle it. Should it affect FontSize property? Or should FontSize remain as is, but TextFormatter should scale it.

And how it's done in WPF.
@Gillibald should have an idea about the best way to implement it.

**Reply by maxkatz6** [reply]
In general, Font scaling is kinda considered a bad practice in UX, when render scaling should be increased instead.
But imo we still need it as an option at least for accessibility.

**Reply by StefanKoell** [reply]
I agree that scaling would be better instead of changing the font size but I needed to go the font size route because changing the scaling is not that easy and straight forward:

- menus are not affected and must be handled explicitly
- popups also
- overlays as well
- there are some quirks with a couple of controls (TreeDataGrid's scroll viewer is affected and misbehaving if you increase scaling)

If there's an easy or easier way to do the scaling approach, I'm all ears ;)

**Reply by timunie** [reply]
I override the dynmic resources accordingly via `App.Current.Resources["MyFontSize"]=Settings.MyFontSize;` 

https://github.com/AvaloniaUI/Avalonia/blob/0c0d4497ef156aa6defaa4cce05fac5831576dd4/src/Avalonia.Themes.Fluent/Accents/BaseResources.xaml#L7

**Comment by Gillibald** [comment]
I have no idea how the system is reporting the text dpi. WPF in general scales text with `PixelsPerDip`
